force-app/main/default/classes/ExtensionControllerTest.cls

```diff
@@ -0,0 +1,144 @@
+
+        //Get the Standard Controller and pass it to controller extension
+        ExtentionController ext = new ExtentionController(accountController);
+        
+        Test.startTest();
+        //Testing
+        ext.setPrimary();
+        
+        Test.stopTest();
+        
+        //Assertions
+        Contact assertCon = [Select LastName, Is_Primary_Contact__c FROM Contact WHERE LastName='TestCon2'];
+        System.assert(assertCon.Is_Primary_Contact__c == true);
+        
+
+    }
+    
+    //Testing search() method
+    @isTest 
+    private static void testsearchSuccess()
+    {   
+        initData('testMethodNameSuccess');
+        
+        
+        List<Contact> Cons = [Select Id, LastName, Is_Primary_Contact__c FROM Contact WHERE LastName Like '%TestCon%'];
+        
+         
+        String sText = 'TestCon';
+        
+        PageReference searchPage = Page.PrimaryContact; 
+               Test.setCurrentPage(searchPage);
+               searchPage.getParameters().put('Id', String.valueOf(refAcct.Id));
+               searchPage.getParameters().put('searchText',sText);
+               
+        //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(refAcct);
+
+        //Get the Standard Controller and pass it to controller extension
+        ExtentionController ext = new ExtentionController(accountController);
+        
+        //setting ContactMap to dummy values
+        ext.extContacts = Cons;
+        
+        //Test
+        Test.startTest();
+        
+        PageReference ref = ext.search();
+        
+        Test.stopTest();
+        
+        //Assert
+        System.assertEquals(4,ext.extContacts.size());
+    }
+    
+    //Test that will fail
+     @isTest
+    private static void testMethodNameFailure() {
+        initData('testMethodNameFailure');
+        
+        
+        String sText = 'TestC';        
+        PageReference failPage = Page.PrimaryContact; 
+               Test.setCurrentPage(failPage);
+               failPage.getParameters().put('Id', String.valueOf(failAct.Id));
+               failPage.getParameters().put('searchText',sText);
+        
+        //create a standard controller and pass the dummy account controller
+        ApexPages.StandardController accountController = new ApexPages.StandardController(failAct);
+
+        //Get the Standard Controller and pass it to controller extension
+        ExtentionController ext = new ExtentionController(accountController);
+        
+        //Setting ContactMap to null
+        ext.extContacts = null;
+        
+        //Feeding with null Map<Integer, Contact>
+        Test.startTest();
+        
+        ext.search();
+        
+        Test.stopTest();
+        
+        //Assertion
+        System.assertEquals(4, ext.extContacts.size());
```

Comment:
I expect in fail scenario to see empty list or smth like that. Right now it looks the same as success scenario